packages/react-core/src/components/TextInputGroup/__tests__/TextInputGroup.test.tsx

```diff
@@ -1,166 +1,99 @@
 import React from 'react';
-import { render } from '@testing-library/react';
-import { mount } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
 import { TextInputGroup, TextInputGroupContext } from '../TextInputGroup';
-import { TextInputGroupMain } from '../TextInputGroupMain';
-import { TextInputGroupUtilities } from '../TextInputGroupUtilities';
-import SearchIcon from '@patternfly/react-icons/dist/esm/icons/search-icon';
 
 describe('TextInputGroup', () => {
-  test('renders content', () => {
-    const view = render(
-      <TextInputGroup>
-        <TextInputGroupMain>Foo</TextInputGroupMain>
-        <TextInputGroupUtilities />
-      </TextInputGroup>
-    );
-    expect(view.container).toMatchSnapshot();
-  });
+  it('renders without children', () => {
+    render(<TextInputGroup data-testid="TextInputGroup" />);
 
-  test('gets custom class and id', () => {
-    const view = render(
-      <TextInputGroup className="custom-class" id="test-id">
-        <TextInputGroupMain>Foo</TextInputGroupMain>
-        <TextInputGroupUtilities />
-      </TextInputGroup>
-    );
-    expect(view.container).toMatchSnapshot();
-  });
+    console.log(screen.getByTestId('TextInputGroup'));
 
-  test('renders with the proper stying when disabled', () => {
-    const view = render(
-      <TextInputGroup isDisabled>
-        <TextInputGroupMain>Foo</TextInputGroupMain>
-        <TextInputGroupUtilities />
-      </TextInputGroup>
-    );
-    expect(view.container).toMatchSnapshot();
-  });
+    expect(screen.getByTestId('TextInputGroup')).toBeVisible();
 
-  test('passes isDisabled=false to children via a context when isDisabled prop is not passed', () => {
-    const TestComponent: React.FunctionComponent = () => {
-      const context = React.useContext(TextInputGroupContext);
+    /** the below expectation is to verify that that the component isn't always rendering with the text used to verify
+    child rendering in the next test */
+    expect(screen.queryByText('Test')).not.toBeInTheDocument();
```

Comment:
For this, would checking that the `children` length is 0 (or something similar) be viable for this test instead?

Reply:
I can get why you would want that, since this expectation only makes sense with the context of the following test, but I feel like checking the length of the `children` property isn't a very "RTL" approach. @jeffpuzzo thoughts?

Reply:
I don't think lines 15 through 17 should be included at all since you're already verifying the parent component is visible without children which is the objective of the test, correct?

Reply:
I partially agree?

15-17 aren't strictly necessary to verify that it renders without children, *but* in the same way that the next test verifies chilldren are rendered (by testing that when "Test" is passed as a child it is rendered to the page), this ensures that the inverse is true (by testing that when "Test" isn't passed it isn't rendered to the page). Essentially it protects from flaws in the test suite, like the component potentially always rendering "Test".

I figure that it goes along with the other aspect of testing that it renders without children for the same reason that checking for "Test" is valid for checking that children are rendered. If you disagree though and think it's either unnecessary or should be a separate test I'm not super committed to it.

Reply:
I do disagree. We're specifically rendering this component with no children (aka a self-closing tag). All we care about is that the component still renders.

Testing that text we're purposefully not including in `render` doesn't exist doesn't make sense to me, as much as it might appear more-so as an inverse of the other test. If someone updates the test to include "Test" as a child after the fact for some reason, then we have bigger problems.

Reply:
I think my concern is that what it's actually testing for doesn't exactly fit with what the intention is. I'd say comparing to a snapshot of the component rendered without children might be better, but that's less an option if we only want the single snapshot.

Reply:
👍 I can understand that/those viewpoints. I'll update shortly.